lib/cmps/base.ts

```diff
@@ -83,12 +83,16 @@ export default class AutoConsentCMPBase implements AutoCMP, DomActionsProvider {
             return false;
         }
 
-        if (runCtx.urlPattern && !window.location.href.match(runCtx.urlPattern)) {
+        if (runCtx.urlPattern && !this.hasMatchingUrlPattern()) {
```

Comment:
Nit: there are some duplicate checks for `urlPattern` here and inside `hasMatchingUrlPattern()`. Is there a way to simplify it?
Maybe make sure that `this.runContext` always has a value? Then we could rely on the check inside `hasMatchingUrlPattern()`, and also simplify some checks in `findCmp()`